cnd/src/http_api/swap_resource.rs

```diff
@@ -107,21 +108,48 @@ pub enum IncludeState {
     No,
 }
 
+/// Build siren entity.  For failed swaps do not return any actions.
+pub fn build_rfc003_siren_entity_no_err<S: StateStore>(
+    state_store: &S,
+    swap: Swap,
+    types: SwapTypes,
+    include_state: IncludeState,
+) -> anyhow::Result<siren::Entity> {
+    build_rfc003_siren_entity(state_store, swap, types, include_state, false)
+}
+
+/// Build siren entity.  For failed swaps return 500 error.
+pub fn build_rfc003_siren_entity_err<S: StateStore>(
```

Comment:
I am trying to think of a better name...
```suggestion
pub fn build_rfc003_siren_entity_err_on_fail<S: StateStore>(
```

Reply:
Use better function name